framework/src/materials/GenericFunctorMaterial.C

```diff
@@ -0,0 +1,84 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GenericFunctorMaterial.h"
+#include "Function.h"
+
+registerMooseObject("MooseApp", GenericFunctorMaterial);
+registerMooseObject("MooseApp", ADGenericFunctorMaterial);
+registerMooseObjectRenamed("MooseApp",
+                           GenericConstantFunctorMaterial,
+                           "06/30/2022 24:00",
+                           GenericFunctorMaterial);
+registerMooseObjectRenamed("MooseApp",
+                           ADGenericConstantFunctorMaterial,
+                           "06/30/2022 24:00",
+                           ADGenericFunctorMaterial);
+registerMooseObjectRenamed("MooseApp",
+                           GenericFunctionFunctorMaterial,
+                           "06/30/2022 24:00",
+                           GenericFunctorMaterial);
+registerMooseObjectRenamed("MooseApp",
+                           ADGenericFunctionFunctorMaterial,
+                           "06/30/2022 24:00",
+                           ADGenericFunctorMaterial);
+
+template <bool is_ad>
+InputParameters
+GenericFunctorMaterialTempl<is_ad>::validParams()
+{
+  InputParameters params = FunctorMaterial::validParams();
+  params += SetupInterface::validParams();
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_LINEAR, EXEC_NONLINEAR};
```

Comment:
I probably mislead/misunderstood you earlier. I think the default for this should be `EXEC_ALWAYS`. Especially for constants, this evaluation should be very cheap. By setting `execute_on` to anything other than `EXEC_ALWAYS` we are committing to caching

Reply:
It looks like you changed this back somehow

Reply:
ugh must have been the force-push